src/lib.rs

```diff
@@ -0,0 +1,420 @@
+        BlogPostChildReplyIds get (fn post_child_reply_ids): map (T::BlogId, T::PostId, T::ReplyId) => Option<BTreeSet<T::ReplyId>>;
+
+        BlogPost get(fn post_by_id): map (T::BlogId, T::PostId) => Option<Post>;
+
+        Reply get (fn reply_by_id): map (T::BlogId, T::PostId, T::ReplyId) => Option<Vec<u8>>;
+
+        BlogLockedStatus get(fn blog_locked): map T::BlogId => bool;
+
+        PostLockedStatus get(fn post_locked): map (T::BlogId, T::PostId) => bool;
+
+        //Reply Ids set, associated with owner
+        ReplyIds get (fn reply_ids_by_owner): map T::AccountId => Option<BTreeSet<(T::BlogId, T::PostId, T::ReplyId)>>;
+
+        BlogsCount get(fn blogs_count): T::BlogId;
+
+        // Overall post count by blog
+        PostsCount get(fn posts_count): map T::BlogId => T::PostId;
+
+        // Overall replies count by blog
+        RepliesCount get(fn replies_count): map (T::BlogId, T::PostId) => T::ReplyId;
+    }
+}
+
+// Blog`s pallet dispatchable functions.
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        // Initializing events
+        fn deposit_event() = default;
+
+        // Security/configuration constraints
+        const POST_TITLE_MAX_LENGTH: MaxLength = T::PostTitleMaxLength::get();
+        const POST_BODY_MAX_LENGTH: MaxLength = T::PostBodyMaxLength::get();
+        const REPLY_MAX_LENGTH: MaxLength = T::ReplyMaxLength::get();
+
+        const POSTS_MAX_NUMBER: MaxNumber = T::PostsMaxNumber::get();
+        const REPLIES_MAX_NUMBER: MaxNumber  = T::RepliesMaxNumber::get();
+        const DIRECT_REPLIES_MAX_NUMBER: MaxNumber = T::DirectRepliesMaxNumber::get();
+
+        const CONSECUTIVE_REPLIES_MAX_NUMBER: MaxConsecutiveRepliesNumber = T::ConsecutiveRepliesMaxNumber::get();
+
+        pub fn create_blog(origin) -> dispatch::Result {
+            let blog_owner = ensure_signed(origin)?;
+            let blogs_count = Self::blogs_count();
+            if <BlogIds<T>>::exists(&blog_owner) {
+                <BlogIds<T>>::mutate(&blog_owner, |blog_ids| {
+                    if let Some(blog_ids) = blog_ids {
+                        blog_ids.insert(blogs_count);
+                    }
+                })
+            } else {
+                let mut new_set = BTreeSet::new();
+                new_set.insert(blogs_count);
+                <BlogIds<T>>::insert(&blog_owner, new_set);
+            }
+            // Blog locking default status
+            <BlogLockedStatus<T>>::insert(blogs_count, false);
+            <BlogsCount<T>>::mutate(|count| *count += T::BlogId::one());
+            Self::deposit_event(RawEvent::BlogCreated(blog_owner, blogs_count));
+            Ok(())
+        }
+
+        pub fn lock_blog(origin, blog_id: T::BlogId) -> dispatch::Result {
+            let blog_owner = ensure_signed(origin)?;
+            ensure!(<BlogIds<T>>::exists(&blog_owner), BLOG_OWNER_NOT_FOUND);
+            match Self::blog_ids_by_owner(&blog_owner) {
```

Comment:
The code flow is slightly hard to read here, as its not trivial to see exactly where it is safe to start mutating.

Try to follow this best practice

https://github.com/Joystream/joystream/issues/36#issuecomment-543174304

Also, try to always write a comment on the specific line number where mutation is safe, sort of like this 

https://github.com/Joystream/substrate-runtime-joystream/blob/master/runtime-modules/content-working-group/src/lib.rs#L1162

If you are able to write this, then you know you are following the best practice link.

Reply:
Nice tip, thank you.

Reply:
Fixed here https://github.com/iorveth/substrate-blog-module/commit/a682cc3630bf28aa110ac0c0186936885c3fc2e3#diff-b4aea3e418ccdb71239b96952d9cddb6R262